<?xml version="1.0"?>
<launch>
  <!-- Input/Output Topics: -->
  <arg name="point_cloud_in" default="/transformed/point_cloud"/>

  <arg name="debug_mode" default="false"/>
  <arg name="start_manager" default="true"/>
  <arg name="nodelet_manager" default="table_nodelet_manager"/>

  <group if="$(arg start_manager)">
    <node if="$(arg debug_mode)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"
          launch-prefix="gdb -ex run --args"/>
    <node unless="$(arg debug_mode)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"/>
  </group>

  <node pkg="nodelet" type="nodelet" name="table_region_crop"
        args="load cob_table_object_cluster/TableRegionCropNodelet $(arg nodelet_manager)" output="screen">

    <remap from="point_cloud_in" to="$(arg point_cloud_in)"/>
    <remap from="point_cloud_out" to="/table_region_crop/point_cloud_out"/>

    <rosparam command="load" file="$(find cob_table_object_cluster)/ros/launch/table_region_crop.yaml"/>
  </node>


</launch>

